src/CSharp/CodeCracker/Refactoring/XmlDocumentationCodeFixProvider.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxKind;
+using System;
+using System.Collections.Generic;
+
+namespace CodeCracker.CSharp.Refactoring
+{
+    public abstract class XmlDocumentationCodeFixProvider : CodeFixProvider
+    {
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(DiagnosticId.XmlDocumentation.ToDiagnosticId());
+
+        public abstract SyntaxNode FixParameters(MethodDeclarationSyntax method, SyntaxNode root);
+
+        public abstract string MessageFormat { get; }
```

Comment:
No need for this property to exist on the base class, it is not used here. It can be a const string field (not a property) on the derived classes